File: Java/821. Shortest Distance to a Character.java
class Solution {
    public int[] shortestToChar(String s, char c) {
        int l = s.length(), m;
        int temp[] = new int[l], res[] = new int[l];

        for(int i = 0; i < l; i++){
            if(s.charAt(i) == c) temp[i] = 1;
        }

        for(int i = 0; i < l; i++){
            m = 1001;
            for(int j = 0; j < l; j++){
                if(temp[j] == 1){
                    m = Math.min(m, Math.abs(j - i));
                }
            }
            res[i] = m;
        }

        return  res;
    }
}

File: Python/821. Shortest Distance to a Character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        temp, res = [], []

        for i, x in enumerate(s):
            if x == c: temp.append(i)
        
        for i, x in enumerate(s):
            m = 1001
            for j in temp: m = min(m, abs(i - j))
            res.append(m)
        
        return res
